src/processor/operator/persistent/index_builder.cpp

```diff
@@ -92,51 +92,68 @@ void IndexBuilderSharedState::quitProducer() {
     }
 }
 
-void IndexBuilder::insert(const ColumnChunkData& chunk, offset_t nodeOffset, offset_t numNodes) {
-    checkNonNullConstraint(chunk.getNullData(), numNodes);
+void IndexBuilder::insert(const ColumnChunkData& chunk, offset_t nodeOffset, offset_t numNodes,
+    IndexBuilderErrorHandler& errors) {
     TypeUtils::visit(
         chunk.getDataType().getPhysicalType(),
         [&]<HashablePrimitive T>(T) {
             for (auto i = 0u; i < numNodes; i++) {
-                auto value = chunk.getValue<T>(i);
-                localBuffers.insert(value, nodeOffset + i);
+                if (checkNonNullConstraint(chunk, nodeOffset, i, errors)) {
+                    auto value = chunk.getValue<T>(i);
+                    localBuffers.insert(value, nodeOffset + i, errors);
+                }
             }
         },
         [&](ku_string_t) {
             auto& stringColumnChunk =
                 ku_dynamic_cast<const ColumnChunkData&, const StringChunkData&>(chunk);
             for (auto i = 0u; i < numNodes; i++) {
-                auto value = stringColumnChunk.getValue<std::string>(i);
-                localBuffers.insert(std::move(value), nodeOffset + i);
+                if (checkNonNullConstraint(chunk, nodeOffset, i, errors)) {
+                    auto value = stringColumnChunk.getValue<std::string>(i);
+                    localBuffers.insert(std::move(value), nodeOffset + i, errors);
+                }
             }
         },
-        [&](auto) {
+        [&]<typename T>(T) {
             throw CopyException(ExceptionMessage::invalidPKType(chunk.getDataType().toString()));
```

Comment:
Don't think ignoring this error makes sense (since it applies to an entire column) so didn't make it skippable